parquet-column/src/main/java/parquet/schema/Types.java

```diff
@@ -417,33 +478,26 @@ protected DecimalMetadata decimalMetadata() {
     }
   }
 
-  /**
-   * A builder for {@link GroupType} objects.
-   *
-   * @param <P> The type that this builder will return from
-   *          {@link #named(String)} when the type is built.
-   */
-  public static class GroupBuilder<P> extends Builder<GroupBuilder<P>, P> {
+  public abstract static class BaseGroupBuilder<P, Q extends BaseGroupBuilder> extends
```

Comment:
I played around with the generics a bit in my local copy and I think I have a slightly cleaner signature for this:

```java
  public abstract static class
      BaseGroupBuilder<P, T extends BaseGroupBuilder<P, T>>
      extends Builder<T, P> { ... }
```

I used T (as in this) for the self type throughout, but it is similar to what you have here. The main thing is that the concrete GroupBuilder types will return their own types from self() for method chaining.